webapp/src/components/github_repo_selector.jsx

```diff
@@ -4,130 +4,67 @@
+            this.setState({repoOptions: options});
         }).catch((e) => {
             this.setState({error: e});
         });
     };
 
-    debouncedSearchRepo = debounce(this.searchRepo, searchDebounceDelay);
-
-    onChange = (e) => {
-        const value = e ? e.value : '';
-        this.props.onChange(value);
-    }
+    onChange = (name, newValue) => {
+        this.props.onChange(newValue);
+    };
 
     render() {
-        const {error} = this.state;
-        const requiredStar = (
-            <span
-                className={'error-text'}
-                style={{marginLeft: '3px'}}
-            >
-                {'*'}
-            </span>
-        );
-
-        let issueError = null;
-        if (error) {
-            issueError = (
-                <p className='help-text error-text'>
-                    <span>{error}</span>
-                </p>
-            );
-        }
-
-        const serverError = this.state.error;
-        let errComponent;
-        if (this.state.error) {
-            errComponent = (
-                <p className='alert alert-danger'>
-                    <i
-                        className='fa fa-warning'
-                        title='Warning Icon'
-                    />
-                    <span>{serverError.toString()}</span>
-                </p>
-            );
-        }
-
-        const requiredMsg = 'This field is required.';
-        let validationError = null;
-        if (this.props.showErrors && this.props.required && this.state.invalid) {
-            validationError = (
-                <p className='help-text error-text'>
-                    <span>{requiredMsg}</span>
-                </p>
-            );
-        }
-
         return (
             <div className={'form-group margin-bottom x3'}>
-                {errComponent}
-                <label
-                    className={'control-label'}
-                    htmlFor={'issue'}
-                >
-                    {'GitHub Repository'}
-                </label>
-                {this.props.required && requiredStar}
-                <AsyncSelect
-                    name={'repository'}
-                    placeholder={'Select repository to post the issue in'}
-                    onChange={this.onChange}
+                <ReactSelectSetting
+                    name={'repo'}
+                    label={'Repository'}
+                    limitOptions={true}
                     required={true}
-                    disabled={false}
+                    onChange={this.onChange}
+                    options={this.state.repoOptions}
                     isMulti={false}
-                    isClearable={true}
-                    defaultOptions={true}
-                    loadOptions={this.handleGetRepoValue}
-                    menuPortalTarget={document.body}
-                    menuPlacement='auto'
-                    styles={getStyleForReactSelect(this.props.theme)}
+                    key={'LT'}
```

Comment:
Doesn't really matter, but idk what `LT` means

```suggestion
                    key={'repo'}
```